src/sentry_plugins/trello/plugin.py

```diff
@@ -0,0 +1,278 @@
+        Pull the value out of our the arguments to this function or from the DB
+        """
+
+        def get_value(field):
+            initial_values = kwargs.get("initial") or {}
+            if initial_values.get(field):
+                return initial_values[field]
+            return self.get_option(field, project)
+
+        token_config = {
+            "name": "token",
+            "type": "secret",
+            "label": "Trello API Token",
+            "default": None,
+        }
+
+        token_val = get_value("token")
+        if token_val:
+            token_config["required"] = False
+            token_config["prefix"] = token_val[:5]
+            token_config["has_saved_value"] = True
+        else:
+            token_config["required"] = True
+
+        key_val = get_value("key")
+
+        key_config = {
+            "name": "key",
+            "type": "text",
+            "required": True,
+            "label": "Trello API Key",
+            "default": key_val,
+        }
+
+        config = [key_config, token_config]
+        org_value = get_value("organization")
+        include_org = kwargs.get("add_additial_fields") or org_value
+        if key_val and token_val and include_org:
+            trello_client = TrelloApiClient(key_val, token_val)
+            try:
+                org_options = trello_client.get_organization_options()
+                config.append(
+                    {
+                        "name": "organization",
+                        "label": "Trello Organization",
+                        "choices": org_options,
+                        "type": "select",
+                        "required": False,
+                        "default": org_value,
+                    }
+                )
+            except Exception as e:
+                self.raise_error(e)
+        return config
+
+    def validate_config(self, project, config, actor=None):
+        """
+        Make sure the configuration is valid by trying to query for the organizations with the auth
+        """
+        trello_client = TrelloApiClient(config["key"], config["token"])
+        try:
+            trello_client.get_organization_options()
+        except Exception as e:
+            raise PluginError(self.message_from_error(e))
+        return config
+
+    def get_group_urls(self):
+        """
+        Return the URLs and the matching views
+        """
+        return super(TrelloPlugin, self).get_group_urls() + [
+            url(
+                r"^options",
+                IssueGroupActionEndpoint.as_view(view_method_name="view_options", plugin=self),
+            ),
+            url(
+                r"^autocomplete",
+                IssueGroupActionEndpoint.as_view(view_method_name="view_autocomplete", plugin=self),
+            ),
+        ]
+
+    def is_configured(self, request, project, **kwargs):
+        return all((self.get_option(key, project) for key in ("token", "key")))
```

Comment:
Can omit a set of parens 😃 Not sure if this is `black` at work tho.

```suggestion
        return all(self.get_option(key, project) for key in ("token", "key"))
```

Reply:
Also, might be nice to have `REQUIRED_CONFIG_KEYS=("token", "key")` and using it here.